players/group_3.py

```diff
@@ -49,15 +57,33 @@ def init_cuts(self, current_percept) -> Tuple[List[List[int]]]:
         ([[y1, y1], [y2, y2]], [[x1, x1], [x2, x2]])          # i.e., (HORIZONTAL_CUTS, VERTICAL_CUTS)
         """
         cake_len, cake_width = current_percept.cake_len, current_percept.cake_width
-        row_count, col_count = 2, 4
-        row_height, col_width = cake_len / row_count, cake_width / col_count
-
-        horizontal_cuts = [[i * row_height, i * row_height] for i in range(1, row_count)]
-        vertical_cuts = [[i * col_width, i * col_width] for i in range(1, col_count)]
 
-        print(f"{cake_len=}, {cake_width=}")
-        print(f"init_{horizontal_cuts=}, {vertical_cuts=}")
-        return horizontal_cuts, vertical_cuts
+        working_pairs = []
+        extra_pieces = 0
+        extra_area = cake_len * cake_width * 0.05
+        while not working_pairs:
+            mod_requests = current_percept.requests
+            mod_requests += [extra_area / extra_pieces for _ in range(extra_pieces)]
+            factor_pairs = self.find_factor_pairs(len(mod_requests))
+
+            for factor_pair in factor_pairs:
+                num_rows, num_cols = factor_pair
+                row_height, col_width = cake_len / num_rows, cake_width / num_cols
+                fits_plate = np.sqrt(row_height**2 + col_width**2) <= 25
+                if fits_plate:
+                    working_pairs.append(factor_pair)
+
+            extra_pieces += 1
+
+        cuts = {}
+        for working_pair in working_pairs:
+            num_rows, num_cols = working_pair
+            row_height, col_width = cake_len / num_rows, cake_width / num_cols
+            horizontal_cuts = [[i * row_height, i * row_height] for i in range(1, num_rows)]
+            vertical_cuts = [[i * col_width, i * col_width] for i in range(1, num_cols)]
+            cuts[working_pair] = (horizontal_cuts, vertical_cuts)
+
+        return cuts
```

Comment:
@eddiechoi00 i'm happy with this! Thank you 👍 

one thought i have in my mind is...
- this seems good for identifying viable candidates for rows/cols
- I might have missed it in the code, but it doesn't seem to propose whether one row-col pair is better than another viable row/col pair.
- is that something that could be included in a later step after this func call? 

i feel like at this point its connected to the binning and while we attempt the binning for the first row/col pair, if we can somehow quantify the error or 'fit' we can compute that for each of the viable row/col count pairs generated from here and use the one that fits the best. Not too aware of how the MILP or other binning strategies work at the moment though.

Reply:
i made it so that rn we can try different combinations given our actual solving algo is fast enough